### ISSUE TITLE
"Incompatible types in assignment" when using boto3-stubs-lite

### ISSUE THREAD

**Issue by ericbn** [youtype/mypy_boto3_builder#177]
**Describe the bug**
Mypy fails with "Incompatible types in assignment" when using boto3-stubs-lite and an assignment with a type annotation as in the documentation.

**To Reproduce**

1. Install `boto3-stubs-lite[sqs]`
2. Run `mypy` on the following code sample

```python
import os

import boto3
from mypy_boto3_sqs.client import SQSClient

QUEUE_URL = os.environ["QUEUE_URL"]
MAX_NUMBER_OF_MESSAGES = 2

sqs: SQSClient = boto3.client("sqs")
response = sqs.receive_message(
    QueueUrl=QUEUE_URL,
    MaxNumberOfMessages=MAX_NUMBER_OF_MESSAGES,
    MessageAttributeNames=["All"],
)
print(response["Messages"])
```

**Actual output**

```
src/sqs.py:9: error: Incompatible types in assignment (expression has type "BaseClient", variable has type "SQSClient")  [assignment]
Found 1 error in 1 file (checked 1 source file)
```

**Expected output**

```
Success: no issues found in 1 source file
```

**Additional context**
Using boto3-stubs-lite==1.26.41 and mypy-boto3-sqs==1.26.0.post1.

Two workarounds for the issue are:
1. `pip uninstall boto3-stubs-lite` (uninstall boto3-stubs-lite or just install mypy-boto3-sqs)
2. `sqs: SQSClient = boto3.client("sqs")  # type: ignore[assignment]` (add the comment)

I was hoping it could work without any of the workarounds.

**Comment by vemel** [comment]
Unfortunately, the only way is to explicitly specify type and `type: ignore` it. I will try to find a better solution.

**Comment by vemel** [comment]
@ericbn  thank you for the report and the fix. I tested it and released a new version with the fix. Fixed `boto3-stubs-lite` should be automatically released tomorrow.

Please try in out and let me know if it works as it should.

**Comment by ericbn** [comment]
Worked, thanks! 🎉
